### PR TITLE
VACMS-13613 and 8017 Add PostFacilityBase methods

### PR DESCRIPTION
**Pull request by swirtSJW**
These methods are not being used yet, but they will be used by two different PRs for Lighthouse that we can not control when they will land.  Adding these here now in andvance helps avoid conflct.

## Description

Relates to  #13613
Relates to #8017 

## Testing done
Code is not being used.  Testing will come when either of the PRs that use it are reviewed.

## QA steps
This code is not being used.  Testing will come when either of the PRs that use it are reviewed.

### CODE REVIEW

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php [hunk 1]
```diff
@@ -222,29 +222,6 @@ protected function getFacilityUrl(): string | null {
...
-      // let's do our best to trim this down if we need to
-      if ($operatingStatusMoreInfoLength > 300) {
-        $operatingStatusMoreInfo = str_replace('&nbsp;', ' ', $operatingStatusMoreInfo);
-        $operatingStatusMoreInfo = trim($operatingStatusMoreInfo);
-        $operatingStatusMoreInfo = preg_replace("/(\r?\n|\r)+/", " ", $operatingStatusMoreInfo);
-      }
-    }
-    return $operatingStatusMoreInfo;
-  }
-
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php:247]: Moved this to Base because VetCenter CAPs are going to need it too.

**Comment by swirtSJW** [comment]
image

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php [hunk 2]
```diff
@@ -116,4 +140,216 @@ protected function isLovellTricareSection(EntityInterface $entity) : bool {
...
+   * @param string $time
+   *   A time including am or pm.
+   *
+   * @return string
+   *   The time passed in or 'noon' or 'midnight'.
+   */
+  protected function midnightNoonify(string $time): string {
+    $time = trim($time);
+    if ($time === '12:00 am' || $time === '12:00 a.m.') {
+      $time = 'noon';
```
Comment (omahane) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php:288]: I think these are backwards.
```suggestion
      $time = 'midnight';
```

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php [hunk 3]
```diff
@@ -116,4 +140,216 @@ protected function isLovellTricareSection(EntityInterface $entity) : bool {
...
+    }
+    elseif ($time === '12:00 pm' || $time === '12:00 p.m.') {
+      $time = 'midnight';
```
Comment (omahane) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php:291]:
```suggestion
      $time = 'noon';
```

**Review by omahane — COMMENTED** [review]
This looks good. 
I did notice something that looks mixed up and noted it.

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php (hunk 2 above)
Comment (swirtSJW) [reply docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php:288]: Great googly moogly.   Good catch.

**Pull request merged**